client/pd_service_discovery.go

```diff
@@ -345,14 +374,29 @@ func (c *pdServiceDiscovery) initClusterID() error {
 	return nil
 }
 
+func (c *pdServiceDiscovery) updateServiceMode() {
+	leaderAddr := c.getLeaderAddr()
+	if len(leaderAddr) > 0 {
+		clusterInfo, err := c.getClusterInfo(c.ctx, leaderAddr, c.option.timeout)
+		if err != nil {
+			log.Warn("[pd] failed to get cluster info for the leader", zap.String("leader-addr", leaderAddr), errs.ZapError(err))
+			return
+		}
+		c.serviceModeUpdateCb(clusterInfo.ServiceModes[0])
```

Comment:
we only use [0]?

Reply:
Yes, for now.

Reply:
Either API or PD. The array is for extensibility in future.